fdbclient/common.go

```diff
@@ -42,43 +42,55 @@ var DefaultCLITimeout = 10 * time.Second
 
 // getFDBDatabase opens an FDB database. The result will be cached for
 // subsequent calls, based on the cluster namespace and name.
-func getFDBDatabase(cluster *fdbv1beta2.FoundationDBCluster) (fdb.Database, error) {
+func getFDBDatabase(cluster *fdbv1beta2.FoundationDBCluster) (fdb.Database, string, error) {
 	clusterFile, err := os.CreateTemp("", "")
 	if err != nil {
-		return fdb.Database{}, err
+		return fdb.Database{}, "", err
 	}
 
 	defer clusterFile.Close()
 	clusterFilePath := clusterFile.Name()
 
 	_, err = clusterFile.WriteString(cluster.Status.ConnectionString)
 	if err != nil {
-		return fdb.Database{}, err
+		return fdb.Database{}, clusterFilePath, err
 	}
 	err = clusterFile.Close()
 	if err != nil {
-		return fdb.Database{}, err
+		return fdb.Database{}, clusterFilePath, err
 	}
 
 	database, err := fdb.OpenDatabase(clusterFilePath)
 	if err != nil {
-		return fdb.Database{}, err
+		return fdb.Database{}, clusterFilePath, err
 	}
 
 	err = database.Options().SetTransactionTimeout(defaultTransactionTimeout.Milliseconds())
 	if err != nil {
-		return fdb.Database{}, err
+		return fdb.Database{}, clusterFilePath, err
 	}
 
-	return database, nil
+	return database, clusterFilePath, nil
 }
 
 func getValueFromDBUsingKey(cluster *fdbv1beta2.FoundationDBCluster, log logr.Logger, fdbKey string, timeout time.Duration) ([]byte, error) {
 	log.Info("Fetch values from FDB", "namespace", cluster.Namespace, "cluster", cluster.Name, "key", fdbKey)
 	defer func() {
 		log.Info("Done fetching values from FDB", "namespace", cluster.Namespace, "cluster", cluster.Name, "key", fdbKey)
 	}()
-	database, err := getFDBDatabase(cluster)
+	database, clusterFilePath, err := getFDBDatabase(cluster)
+	// Ensure we delete the temporary file
+	defer func() {
+		if clusterFilePath == "" {
+			return
+		}
+
+		err = os.Remove(clusterFilePath)
```

Comment:
Does removing the file clean up the client cache maintained by the bindings, or will we still be leaking all of those resources? I think that cache is keyed off of the cluster file path, so having the cluster file deleted and potentially reused later could produce additional undesired behavior.

Reply:
reference on the caching in the go bindings: https://github.com/apple/foundationdb/blob/main/bindings/go/src/fdb/fdb.go#L283

Reply:
I think leaking the cache maintained by the bindings while removing the file is more dangerous than the current behavior. Without changes in the FDB client itself, I think the best we can do is to make the cluster file path deterministic. We can base it on the UUID of the FoundationDBCluster object so we can gracefully handle clusters being deleted and recreated with the same cluster name but different connection strings and volumes.